src/ApiService/ApiService/onefuzzlib/NsgOperations.cs

```diff
@@ -3,24 +3,24 @@
 
 
 namespace Microsoft.OneFuzz.Service {
-    public interface INsg {
+    public interface INsgOperations {
         Async.Task<NetworkSecurityGroupResource?> GetNsg(string name);
         public Async.Task<Error?> AssociateSubnet(string name, VirtualNetworkResource vnet, SubnetResource subnet);
         IAsyncEnumerable<NetworkSecurityGroupResource> ListNsgs();
         bool OkToDelete(HashSet<string> active_regions, string nsg_region, string nsg_name);
         Async.Task<bool> StartDeleteNsg(string name);
 
-        Async.Task DissociateNic(NetworkInterfaceResource nic);
+        Async.Task<ResultVoid<Error>> DissociateNic(Nsg nsg, NetworkInterfaceResource nic);
     }
 
 
-    public class Nsg : INsg {
+    public class NsgOperations : INsgOperations {
```

Comment:
Renamed `Nsg` -> `NsgOperations` to make room to create a record named `Nsg` to follow our convention.